net-analyzer/linkchecker/linkchecker-9999.ebuild

```diff
@@ -15,44 +17,70 @@ if [[ "${PV}" == "9999" ]]; then
 	EGIT_REPO_URI="https://github.com/linkcheck/linkchecker.git"
 	inherit git-r3
 else
-	SRC_URI=""
+	SRC_URI="https://github.com/linkchecker/linkchecker/archive/refs/tags/v${PV}.tar.gz -> ${P}.gh.tar.gz"
 	KEYWORDS="~amd64 ~x86"
 fi
 
 LICENSE="GPL-2"
 SLOT="0"
-IUSE="sqlite"
-# requires libs not present in portage yet
-RESTRICT="test"
+IUSE="bash-completion nls pdf sqlite"
 
 RDEPEND="
-	dev-python/beautifulsoup4[${PYTHON_USEDEP}]
-	dev-python/dnspython[${PYTHON_USEDEP}]
-	dev-python/pyxdg[${PYTHON_USEDEP}]
-	dev-python/requests[${PYTHON_USEDEP}]
-	dev-python/setuptools_scm[${PYTHON_USEDEP}]
+	$(python_gen_cond_dep '
+		dev-python/beautifulsoup4[${PYTHON_USEDEP}]
+		dev-python/dnspython[${PYTHON_USEDEP}]
+		dev-python/requests[${PYTHON_USEDEP}]
+	')
+	bash-completion? (
+		$(python_gen_cond_dep 'dev-python/argcomplete[${PYTHON_USEDEP}]') )
+	pdf? (
+		$(python_gen_cond_dep 'app-text/pdfminer[${PYTHON_USEDEP}]') )
+"
+BDEPEND="
+	$(python_gen_cond_dep '
+		dev-python/hatch-vcs[${PYTHON_USEDEP}]
+	')
+	nls? (
+		$(python_gen_cond_dep '
+			dev-python/polib[${PYTHON_USEDEP}]
+		')
+	)
+	test? (
+		$(python_gen_cond_dep '
+			app-text/pdfminer[${PYTHON_USEDEP}]
+			dev-python/parameterized[${PYTHON_USEDEP}]
+			dev-python/pyftpdlib[${PYTHON_USEDEP}]
+			dev-python/pyopenssl[${PYTHON_USEDEP}]
+			dev-python/pytest-xdist[${PYTHON_USEDEP}]
+		')
+	)
 "
 
-PATCHES=(
-	"${FILESDIR}/${PN}-9.3-bash-completion.patch"
-)
+distutils_enable_tests pytest
+
+PATCHES=( "${FILESDIR}/${PN}-9.3-bash-completion.patch" )
 
 DOCS=(
 	doc/changelog.txt
 	doc/upgrading.txt
 )
 
-python_prepare_all() {
-	distutils-r1_python_prepare_all
+python_test() {
+	# Telnet test uses miniboa which is not available in Gentoo
+	rm -f tests/checker/test_telnet.py
+	# Multiple warnings about unclosed test sockets with epytest
+	pytest -vra -n "$(makeopts_jobs)"
 }
 
 python_install_all() {
 	distutils-r1_python_install_all
-	newbashcomp config/linkchecker-completion ${PN}
+	if use bash-completion; then
```

Comment:
argcomplete should probably remain as an optfeature, see also https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0301.